include/range/v3/view/subrange.hpp

```diff
@@ -0,0 +1,428 @@
+                    ConvertibleTo<sentinel_t<R>, S>())>
+            constexpr subrange(R &&r)
+              : subrange{ranges::begin(r), ranges::end(r)}
+            {}
+
+            template<typename R,
+                CONCEPT_REQUIRES_(StoreSize &&
+                    ReferenceableRange<R>() &&
+                    SizedRange<R>() &&
+                    ConvertibleTo<iterator_t<R>, I>() &&
+                    ConvertibleTo<sentinel_t<R>, S>())>
+            constexpr subrange(R &&r)
+              : subrange{ranges::begin(r), ranges::end(r), ranges::distance(r)}
+            {}
+
+            template<typename R,
+                CONCEPT_REQUIRES_(K == subrange_kind::sized &&
+                    ReferenceableRange<R>() &&
+                    ConvertibleTo<iterator_t<R>, I>() &&
+                    ConvertibleTo<sentinel_t<R>, S>())>
+            constexpr subrange(R &&r, difference_type_t<I> n)
+              : subrange{ranges::begin(r), ranges::end(r),
+                    (RANGES_ASSERT(!SizedRange<R>() || n == ranges::distance(r)), n)}
+            {}
+
+            template<class PairLike,
+                CONCEPT_REQUIRES_(!StoreSize &&
+                    detail::PairLikeConvertibleTo<PairLike, I, S>())>
+            constexpr subrange(PairLike &&r)
+            : subrange{std::get<0>(static_cast<PairLike &&>(r)),
+                std::get<1>(static_cast<PairLike &&>(r))}
+            {}
+
+            template<class PairLike,
+                CONCEPT_REQUIRES_(K == subrange_kind::sized &&
+                    detail::PairLikeConvertibleTo<PairLike, I, S>())>
+            constexpr subrange(PairLike &&r, difference_type_t<I> n)
+            : subrange{std::get<0>(static_cast<PairLike &&>(r)),
+                std::get<1>(static_cast<PairLike &&>(r)), n}
+            {}
+
+            template<typename PairLike,
+                CONCEPT_REQUIRES_(!Range<PairLike>() &&
+                    detail::PairLikeConvertibleFrom<PairLike, I const &, S const &>())>
+            constexpr operator PairLike() const
+            {
+                return PairLike(first_(), last_());
+            }
+
+            constexpr I begin() const
+                noexcept(std::is_nothrow_copy_constructible<I>::value)
+            {
+                return first_();
+            }
+            constexpr S end() const
+                noexcept(std::is_nothrow_copy_constructible<S>::value)
+            {
+                return last_();
+            }
+            constexpr bool empty() const
+            {
+                return first_() == last_();
+            }
+
+            CONCEPT_REQUIRES(K == subrange_kind::sized)
+            RANGES_CXX14_CONSTEXPR difference_type_t<I> size() const
+            {
+#if RANGES_CXX_IF_CONSTEXPR >= RANGES_CXX_IF_CONSTEXPR_17
+                if constexpr (StoreSize) {
+                    return size_();
+                } else {
+                    return last_() - first_();
+                }
```

Comment:
Brace formatting here, and elsewhere. https://en.wikipedia.org/wiki/Indentation_style#Allman_style